src/components/content/order/serviceActions/ServiceActionParameter.tsx

```diff
@@ -0,0 +1,89 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ * SPDX-FileCopyrightText: Huawei Inc.
+ */
+
+import { UseMutationResult } from '@tanstack/react-query';
+import { Button, Form, Input } from 'antd';
+import React from 'react';
+import '../../../../styles/app.module.css';
+import serviceOrderStyles from '../../../../styles/service-order.module.css';
+import {
+    CreateServiceActionData,
+    ServiceActionRequest,
+    ServiceChangeParameter,
+    ServiceOrder,
+} from '../../../../xpanse-api/generated';
+
+export const ServiceActionParameter = ({
+    serviceId,
+    actionName,
+    actionParameters,
+    createServiceActionRequest,
+}: {
+    serviceId: string;
+    actionName: string;
+    actionParameters: ServiceChangeParameter[] | undefined;
+    createServiceActionRequest: UseMutationResult<ServiceOrder, Error, CreateServiceActionData>;
+}): React.JSX.Element => {
+    const [form] = Form.useForm();
+    const onSubmit = () => {
+        const fieldsValues = form.getFieldsValue() as Record<string, string | number | undefined>;
+        const actionParameters: Record<string, unknown> = {};
+        Object.entries(fieldsValues).forEach(([key, value]) => {
+            actionParameters[key] = value;
+        });
+
+        const request: ServiceActionRequest = {
+            actionName: actionName,
+            actionParameters: actionParameters,
+        };
+        const data: CreateServiceActionData = {
+            serviceId: serviceId,
+            requestBody: request,
+        };
+        createServiceActionRequest.mutate(data);
+    };
+
+    return actionParameters ? (
+        <Form
+            form={form}
+            layout='vertical'
+            onFinish={onSubmit}
+            autoComplete='off'
+            validateTrigger={['onSubmit']}
+            key='createServiceAction'
+            disabled={createServiceActionRequest.isSuccess}
+        >
+            {actionParameters.map((actionParameter) => (
+                <Form.Item
+                    key={actionParameter.name}
+                    name={actionParameter.name}
+                    label={actionParameter.name}
+                    rules={[{ required: true }, { type: actionParameter.dataType }]}
+                    colon={true}
+                    initialValue={
+                        actionParameter.dataType === 'number'
+                            ? Number(actionParameter.initialValue)
+                            : actionParameter.initialValue
+                    }
+                    className={serviceOrderStyles.orderParamsFirstParam}
+                >
+                    <Input
+                        name={actionParameter.name}
+                        showCount
+                        maxLength={256}
```

Comment:
here the input must not have fixed length. Ths must be read from the actin parameter configuration. It is also not just text input always. we must support exactly all types  like dropdown, radio button, text box like in the new order.

I hope the same is also implemented in serviceconfiguration. If not, even that form must be changed.